android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java

```diff
@@ -193,9 +198,22 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
     }
 
     characteristic = gatt.getService(SERVICE_UUID).getCharacteristic(CHARACTERISTIC_WEBPAGE_UUID);
+    checkWebpageGzipped();
+    
     gatt.readCharacteristic(characteristic);
   }
 
+  private void checkWebpageGzipped() {
```

Comment:
it uses the same characteristic. The descriptor is nested within the characteristic.
Check the content of the file to see if it is gzipped? Is there an easy way to do that?